webui-src/app/people/people_ownids.js

```diff
@@ -1,213 +1,399 @@
+                  : "An error occured while creating identity.";
+                widget.popupMessage([
+                  m("h3", "Create new Identity"),
+                  m("hr"),
+                  message,
+                ]);
+              });
+          },
+        },
+        "Enter"
+      ),
+    ],
+  };
+};
 const CreateIdentity = () => {
   // TODO: set user avatar
-  let name = '',
-    pgpPassword = '',
+  let name = "",
+    id = "",
+    pgpPassword = "",
     pseudonimous = false;
   return {
-    view: () => [
-      m('i.fas.fa-user-plus'),
-      m('h3', 'Create new Identity'),
-      m('hr'),
-      m('input[type=text][placeholder=Name]', {
+    view: (v) => [
+      m("i.fas.fa-user-plus"),
+      m("h3", "Create new Identity"),
+      m("hr"),
+      m("input[type=text][placeholder=Name]", {
         value: name,
-        oninput: e => (name = e.target.value),
+        oninput: (e) => (name = e.target.value),
       }),
-      'Type:',
       m(
-        'select',
+        "div",
         {
-          value: pseudonimous,
-          oninput: e => (type = e.target.value),
+          style: "display:inline; margin-left:5px;",
         },
         [
-          m('option[value=false]', 'Linked to your Profile'),
-          m('option[value=true]', 'Pseudonymous'),
-        ],
+          "Type:",
+          m(
+            "select",
+            {
+              value: pseudonimous,
+              style: "border:1px solid black",
+              oninput: (e) => {
+                pseudonimous = e.target.value === "true" ? true : false;
+                console.log(pseudonimous);
+              },
+            },
+            [
+              m("option[value=false][selected]", "Linked to your Profile"),
+              m("option[value=true]", "Pseudonymous"),
+            ]
+          ),
+        ]
       ),
-      m('br'),
-      m('input[type=password][placeholder=Retroshare Passphrase]', {
+      m("br"),
+      m("input[type=password][placeholder=Retroshare Passphrase]", {
         value: pgpPassword,
-        oninput: e => (pgpPassword = e.target.value),
+        oninput: (e) => (pgpPassword = e.target.value),
       }),
       m(
-        'p',
-        'You can have one or more identities. ' +
-          'They are used when you chat in lobbies, ' +
-          'forums and channel comments. ' +
-          'They act as the destination for distant chat and ' +
-          'the Retroshare distant mail system.',
+        "p",
+        "You can have one or more identities. " +
+          "They are used when you chat in lobbies, " +
+          "forums and channel comments. " +
+          "They act as the destination for distant chat and " +
+          "the Retroshare distant mail system."
       ),
       m(
-        'button',
+        "button",
+        {
+          onclick: () => {
+            !pseudonimous
+              ? widget.popupMessage(
+                  m(SignedIdentiy, {name: name, pseudonimous: pseudonimous})
```

Comment:
Why do we have to pass `pseudonimous` here? Since it is a signed identiy, pseudonimous will always be false right?

Reply:
Actually if we are going this way, how about moving the below logic also to inside `SignedIdentity`? We can check for `pseudonimous` there and ask the passphrase depending on that. The "successfully created" message will also not be duplicated.

Reply:
Even after putting the `pseudonimous` check inside the signed identity, the Successfully created message will be duplicated.